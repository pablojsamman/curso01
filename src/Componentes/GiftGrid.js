import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

const GiftGrid = ({categoria}) => {
    
    const {data, loading} = useFetchGifs(categoria)
    
    return (
    <>
        <h3> {categoria} </h3>
        {loading&&<p>Cargando...</p>}


        <div className='card-grid'>

            {
                data.map( img => {
                    return <GifGridItem key={img.id} { ...img }  />
                })
            }
        </div>
    </>
    )
}

GiftGrid.propTypes = {
    categoria: PropTypes.string.isRequired,
}

export default GiftGrid
