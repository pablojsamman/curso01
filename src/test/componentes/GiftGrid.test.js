import React from 'react'
import {shallow} from 'enzyme'
import GiftGrid from '../../Componentes/GiftGrid'

describe('Pruebas en <GifGrid>', () => {
    const categoria = 'Alguna categoria'
    const mockUseFetchGifs = useFetchGifs

    test('debe mostrar componente correctamente', () => {
        
        const wrapper = shallow( <GiftGrid categoria={categoria} /> )
        expect( wrapper ).toMatchSnapshot()
    })

    test('mostrar items cuando carga iamgenes', () => {

        const wrapper = shallow( <GiftGrid categoria={categoria} /> )
        
    })
    
    
})
