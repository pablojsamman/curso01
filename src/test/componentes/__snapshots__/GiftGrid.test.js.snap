// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid> debe mostrar componente correctamente 1`] = `
<Fragment>
  <h3>
     
    Alguna categoria
     
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
